metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/MetadataChangeLogProcessor.java

```diff
@@ -41,61 +43,159 @@
 @EnableKafka
 public class MetadataChangeLogProcessor {
 
-  private final List<MetadataChangeLogHook> hooks;
   private final Histogram kafkaLagStats = MetricUtils.get().histogram(MetricRegistry.name(this.getClass(), "kafkaLag"));
 
+  private static final String GROUP_1 = "GROUP_1";
+  private static final String GROUP_2 = "GROUP_2";
+  private static final String GROUP_3 = "GROUP_3";
+  private static final String GROUP_4 = "GROUP_4";
+
+  private static final String HOOK_NAME_1 = "updateIndicesHook";
+  private static final String HOOK_NAME_2 = "ingestionSchedulerHook";
+  private static final String HOOK_NAME_3 = "entityChangeEventHook";
+  private static final String HOOK_NAME_4 = "siblingAssociationHook";
+
+  @Value("${kafka.mae.consumer.group.1:}")
+  private String consumerGroup1Hooks;
+
+  @Value("${kafka.mae.consumer.group.2:}")
+  private String consumerGroup2Hooks;
+
+  @Value("${kafka.mae.consumer.group.3:}")
+  private String consumerGroup3Hooks;
+
+  @Value("${kafka.mae.consumer.group.4:}")
+  private String consumerGroup4Hooks;
+
+  private final Map<String, List<MetadataChangeLogHook>> groupMap;
+
   @Autowired
   public MetadataChangeLogProcessor(
       @Nonnull final UpdateIndicesHook updateIndicesHook,
       @Nonnull final IngestionSchedulerHook ingestionSchedulerHook,
       @Nonnull final EntityChangeEventGeneratorHook entityChangeEventHook,
       @Nonnull final SiblingAssociationHook siblingAssociationHook
   ) {
-    this.hooks = ImmutableList.of(updateIndicesHook, ingestionSchedulerHook, entityChangeEventHook, siblingAssociationHook);
-    this.hooks.forEach(MetadataChangeLogHook::init);
+    Map<Class, MetadataChangeLogHook> hookMap = new HashMap<>();
+    hookMap.put(updateIndicesHook.getClass(), updateIndicesHook);
+    hookMap.put(ingestionSchedulerHook.getClass(), ingestionSchedulerHook);
+    hookMap.put(entityChangeEventHook.getClass(), entityChangeEventHook);
+    hookMap.put(siblingAssociationHook.getClass(), siblingAssociationHook);
+    hookMap.values().forEach(MetadataChangeLogHook::init);
+
+    groupMap = new HashMap<>();
+    groupMap.put(GROUP_1, getHooksFromConfiguration(consumerGroup1Hooks, hookMap));
+    groupMap.put(GROUP_1, getHooksFromConfiguration(consumerGroup2Hooks, hookMap));
+    groupMap.put(GROUP_1, getHooksFromConfiguration(consumerGroup3Hooks, hookMap));
+    groupMap.put(GROUP_1, getHooksFromConfiguration(consumerGroup4Hooks, hookMap));
+  }
+
+  @KafkaListener(id = "${METADATA_CHANGE_LOG_KAFKA_UPDATE_INDICES_CONSUMER_GROUP_ID:"
+          + "generic-mae-consumer-job-client-update-indices-hook}", topics = {  // TODO: MOZDA DA SE PROMENI OVO ID GRUPE
```

Comment:
What's this hanging TODO comment? Not sure if the translation is correct, but seems related to the group ID changing?